onnxruntime/src/session.rs

```diff
@@ -111,6 +111,41 @@ impl<'a> SessionBuilder<'a> {
         Ok(self)
     }
 
+    /// Options settings to match the Ampere AIO examples. So far, the important one
+    /// is selecting the execution provider.
+    #[cfg(feature = "aio")]
+    pub fn with_aio_settings(self) -> Result<SessionBuilder<'a>> {
+        let mut ptr: *mut *mut ::std::os::raw::c_char = std::ptr::null_mut();
+        let mut len: i32 = 0;
+        unsafe {
+            g_ort().GetAvailableProviders.unwrap()(&mut ptr, &mut len);
```

Comment:
can we propagate these errors? any need to?

Reply:
I don't know? I can't tell if there is an actual failure mode here or not. It's definitely prototype quality because things like the device ID are hard coded to 1 because... that's the value that doesn't crash. I haven't been able to get clarity on whether that will always be true or if there is a way to figure it out. But highlighting it is appreciated, we'll have to look at it again if we move forward. And another +1 to Rust... unwrap() makes code smell easier to see.